src/vs/workbench/parts/files/common/workingFilesModel.ts

```diff
@@ -300,6 +303,25 @@ export class WorkingFilesModel implements IWorkingFilesModel {
 		return null;
 	}
 
+	public restoreRecentlyRemovedEntry(): WorkingFileEntry {
```

Comment:
I had a go at handling deleted files, basically this is what it tries to do:

```
if all files exist
  open them
else
  if list is empty
    run action again
  else
    open existing files only
```

The code is a little messier than the above because I wanted to handle non-ENOENT errors too.